### PR TITLE
Bugfix C-extension instrospection.

### PR DESCRIPTION
**Pull request by tristanlatr**
This PR replaces PR #378. 

I've addded tests to what @ntamas had already coded. 
This fixes a bug, So I think I'll release a new version of pydoctor once this is merged

### CODE REVIEW

**Comment by ntamas** [comment]
Thank you for your hard work on this, and also for finishing what I've started! Looking forward to seeing this in production.

File: pydoctor/test/testpackages/c_module_invalid_text_signature/setup.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from setuptools import setup, Extension
+
+cmodule = Extension("mymod.base", sources=["mymod/base.c"],)
+setup(
+    name="mymod",
+    ext_modules=[cmodule],
+    packages=['mymod'],
+)
```
Comment (adiroiban) [inline pydoctor/test/testpackages/c_module_invalid_text_signature/setup.py:8]: a new line will be nice for future clean diffs ...

**Review by adiroiban — APPROVED** [review]
All good. Thanks. Looking forward to the new release.

File: pydoctor/test/testpackages/c_module_invalid_text_signature/setup.py (hunk 1 above)
Comment (tristanlatr) [inline pydoctor/test/testpackages/c_module_invalid_text_signature/setup.py:8]:
```suggestion
)

```

**Pull request merged**